testsuite/tests/typing-recmod/pr6491.ml

```diff
@@ -0,0 +1,27 @@
+(* TEST
+   * expect
+*)
+
+module rec Foo : sig class type c = object method x : int end end = Foo
+and Bar : sig class type c = object inherit Foo.c end end = Bar
+and Baz : sig class type c = object inherit Bar.c end end = Baz;;
+[%%expect {|
+Line 2, characters 44-49:
+2 | and Bar : sig class type c = object inherit Foo.c end end = Bar
+                                                ^^^^^
+Error: Illegal recursive module reference
```

Comment:
This is not a particularly clear error message.  Could the error explain that the problem is specifically due to classes?